Client/Frontend/Autofill/CreditCard/CreditCardSettingsView/CreditCardSettingsViewController.swift

```diff
@@ -0,0 +1,146 @@
+            viewModel: viewModel.addEditViewModel,
+            removeButtonColor: Color(themeManager.currentTheme.colors.textWarning),
+            borderColor: Color(themeManager.currentTheme.colors.borderPrimary)))
+        self.creditCardTableViewController = CreditCardTableViewController(viewModel: viewModel.creditCardTableViewModel)
+        self.notificationCenter = notificationCenter
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("not implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        viewSetup()
+        listenForThemeChange()
+        applyTheme()
+    }
+
+    func viewSetup() {
+        guard let emptyCreditCardView = creditCardEmptyView.view,
+            let addEditCreditCardView = creditCardAddEditView.view,
+            let creditCardTableView = creditCardTableViewController.view else { return }
+        creditCardTableView.translatesAutoresizingMaskIntoConstraints = false
+        emptyCreditCardView.translatesAutoresizingMaskIntoConstraints = false
+        addEditCreditCardView.translatesAutoresizingMaskIntoConstraints = false
+
+        addChild(creditCardEmptyView)
+        addChild(creditCardAddEditView)
+        addChild(creditCardTableViewController)
+        view.addSubview(emptyCreditCardView)
+        view.addSubview(addEditCreditCardView)
+        view.addSubview(creditCardTableView)
+
+        NSLayoutConstraint.activate([
+            emptyCreditCardView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            emptyCreditCardView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+            emptyCreditCardView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            emptyCreditCardView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+
+            addEditCreditCardView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            addEditCreditCardView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+            addEditCreditCardView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            addEditCreditCardView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+
+            creditCardTableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            creditCardTableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+            creditCardTableView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            creditCardTableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor)
+        ])
+
+        // Hide all the views initially until we update the state
+        hideAllViews()
+
+        // Setup state and update view
+        setupState()
+    }
+
+    func setupState() {
+        // check if there are any starting config
+        guard startingConfig != nil else {
```

Comment:
nit: I think this function will be easier to read if we invert the condition of the first guard because the bulk of the work is done inside the guard something like:
```suggestion
        guard startingConfig == nil else {
            updateState(type: .empty)
            return
        }
```